Dependency cleanup -> [storefront] storefront-config -> CheckoutConfig

Solution: For now, we remove CheckoutConfig

Another effort is in progress will replace this with  type augmentation, see GH-10567

```
projects/storefrontlib/src/storefront-config.ts:17:32 - error TS2307: Cannot find module './cms-components/checkout/config/checkout-config' or its corresponding type declarations.
17 import { CheckoutConfig } from './cms-components/checkout/config/checkout-config';

```

--- a/projects/storefrontlib/src/storefront-config.ts
+++ b/projects/storefrontlib/src/storefront-config.ts
@@ -14,7 +14,6 @@ import {
   SiteContextConfig,
   StateConfig,
 } from '@spartacus/core';
-import { CheckoutConfig } from './cms-components/checkout/config/checkout-config';
 import { IconConfig } from './cms-components/misc/icon/icon.model';
 import { QualtricsConfig } from './cms-components/misc/qualtrics/config/qualtrics-config';
 import { PWAModuleConfig } from './cms-structure/pwa/pwa.module-config';
@@ -43,7 +42,6 @@ export type StorefrontConfig =
   | I18nConfig
   | PersonalizationConfig
   | IconConfig
-  | CheckoutConfig
   | GlobalMessageConfig
   | ExternalRoutesConfig
   | ViewConfig

